src/content/language/macros.md

```diff
@@ -0,0 +1,287 @@
+---
+title: Macros (experimental)
+description: Learn about the experimental macros feature as it develops.
+---
+
+[The Dart macro system][spec] is a major new language feature
+***currently under development*** which adds support for
+[static meta-programming][motivation] to the Dart language.
+
+A Dart macro is a user-defineable piece of code that takes in other code as parameters
+and operates on it during compile time to create, modify, or add declarations.
+
+You can think about the macro system in two parts: using macros and writing macros.
+This page covers each (at a high level, as ***the feature is still under development***)
+in the following sections:
+
+- [**The `JsonCodable` macro**](#the-jsoncodable-macro):
+A ready-made macro you can try out today (behind an experiment flag)
+that offers a seamless solution to the
+common issue of tedious JSON serialization in Dart.
+
+- [**The macros feature in general**](#the-macros-language-feature):
+Why we're adding macros to Dart, motivating use cases,
+benefits over existing code gen solutions,
+and a cursory overview of how writing macros works.
+
+[spec]: {{site.repo.dart.lang}}/blob/main/working/macros/feature-specification.md
+[motivation]: {{site.repo.dart.lang}}/language/blob/main/working/macros/motivation.md
+
+## The `JsonCodable` macro
+
+:::important
+The `JsonCodable` macro is not stable and currently behind an [experiment flag][].
+Functionality is subject to change.
+:::
+
+The [`JsonCodable`][] macro encodes and decodes
+user-defined Dart classes to JSON maps of type `Map<String, Object?>`.
+It generates two members, a `toJson` serialization method,
+and a `fromJson` deserialization constructor.
+
+[experiment flag]: /tools/experiment-flags
+[`JsonCodable`]: {{site.repo.dart.sdk}}/tree/main/pkg/json
+
+### Set up the experiment
+
```

Comment:
Maybe mention again here to use dev or beta channel of 3.5 ?

Reply:
Fwiw we don't have a beta release planned afaik, not sure when that will be. Dev should be more up to date and give a better experience right now.

Reply:
(there are lots of fixes that landed last week and a couple still pending also that we will want people to pick up quickly)

Reply:
> Fwiw we don't have a beta release planned afaik, not sure when that will be. Dev (or main) should be more up to date

Does that also apply to [this comment](https://github.com/dart-lang/site-www/pull/5802/files#r1598196483)? (i.e. don't say "beta channel" there either?)

Reply:
I just got confirmation that there won't be beta, so yeah we should say "dev channel" everywhere for macros

Reply:
Ok, I added step 1 under "Set up the experiment"

> 1. Switch to the [dev channel][] and upgrade to at least 3.5.0.